### ISSUE TITLE
Recieved Error: Command failed with exit code 65

### ISSUE THREAD

**Issue by Darrow8** [apache/cordova-ios#855]
# Bug Report
I ran the command `cordova build ios` and it returned `xcodebuild: Command failed with exit code 65`

## Problem
The command failed.

### What is expected to happen?
It should have built correctly.

### What does actually happen?
It doesn't build and returns an error. See the error in the post I made: https://stackoverflow.com/questions/61847125/ionic-cordova-build-ending-with-xcodebuild-command-failed-with-exit-code-65

## Information

I have a full log of what happened here: [ERROR](https://stackoverflow.com/questions/61847125/ionic-cordova-build-ending-with-xcodebuild-command-failed-with-exit-code-65)

### Command or Code

`cordova build ios`

### Environment, Platform, Device

I am running this on Ionic 5 Angular 9. I am on a Macbook Pro Mojave 10.14.6 and Xcode 11.3.1.

### Version information

I included that information in the previous section.

## Checklist

- [x] I searched for existing GitHub issues
- [x] I updated all Cordova tooling to most recent version
- [x] I included all the necessary information above

**Comment by yktan8** [comment]
Having same error. I am on Catalina 10.15.4 and XCode 11.4.1.

Error that I received:
```
** BUILD FAILED **

The following build commands failed:
	PhaseScriptExecution [CP]\ Copy\ Pods\ Resources /Users/myuser/Library/Developer/Xcode/DerivedData/MyApp-eqembrkwgrftfqecxktondfbadbc/Build/Intermediates.noindex/MyApp.build/Debug-iphonesimulator/MyApp.build/Script-C36BD4BBA7C93796DD655B12.sh
(1 failure)
xcodebuild: Command failed with exit code 65

**Comment by maxymczech** [comment]
I have updated to `cordova-ios@6.0.0` and I get this error for `cordova-plugin-inappbrowser`, opened an issue in the plugin repo...

**Comment by marcelmedina** [comment]
I'm getting a consistent error 65, that starts with this error:

```
CompileStoryboard sample/CDVLaunchScreen.storyboard
    cd /Users/mm/Documents/sample/sample.PGK/Core/sample.PGK.Core.Ionic/platforms/ios
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/share/dotnet:~/.dotnet/tools:/Library/Apple/usr/bin:/Library/Frameworks/Mono.framework/Versions/Current/Commands"
    export XCODE_DEVELOPER_USR_PATH=/Applications/Xcode.app/Contents/Developer/usr/bin/..
    /Applications/Xcode.app/Contents/Developer/usr/bin/ibtool --errors --warnings --notices --module sample --output-partial-info-plist /Users/mm/Library/Developer/Xcode/DerivedData/sample-bvrieshnghaqwbegvatmqkxrcaky/Build/Intermediates.noindex/sample.build/Debug-iphonesimulator/sample.build/CDVLaunchScreen-SBPartialInfo.plist --auto-activate-custom-fonts --target-device iphone --target-device ipad --minimum-deployment-target 10.0 --output-format human-readable-text --compilation-directory /Users/mm/Library/Developer/Xcode/DerivedData/sample-bvrieshnghaqwbegvatmqkxrcaky/Build/Intermediates.noindex/sample.build/Debug-iphonesimulator/sample.build /Users/mm/Documents/sample/sample.PGK/Core/sample.PGK.Core.Ionic/platforms/ios/sample/CDVLaunchScreen.storyboard
/* com.apple.ibtool.document.errors */
/Users/mm/Documents/sample/sample.PGK/Core/sample.PGK.Core.Ionic/platforms/ios/sample/CDVLaunchScreen.storyboard:Ze5-6b-2t3: error: Named colors do not work prior to iOS 11.0.
```

In the end I get:
```
** BUILD FAILED **


The following build commands failed:
	CompileStoryboard PlayGoKids/CDVLaunchScreen.storyboard
(1 failure)
xcodebuild: Command failed with exit code 65
```

**Comment by dpogue** [comment]
@marcelmedina 

```
error: Named colors do not work prior to iOS 11.0.
```

If you're using Cordova-iOS 6.x, the minimum supported iOS version is 11.0. You may need to remove a `deployment-target` preference from config.xml if it is set to a lower version.

**Comment by lukas2** [comment]
I see the same problem. In my case the "Builds for" setting in the File Inspector for `CDVLaunchScreen.storyboard` shows a wrong value.

<img width="280" alt="builds_for" src="https://user-images.githubusercontent.com/213249/87165325-4846c000-c2ca-11ea-8e86-1960cebb9a98.png">

In the xml it contains

```
<dependencies>
    <deployment identifier="iOS"/>
    <plugIn identifier="com.apple.InterfaceBuilder.IBCocoaTouchPlugin" version="15706"/>
     <capability name="Named colors" minToolsVersion="9.0"/>
    <capability name="documents saved in the Xcode 8 format" minToolsVersion="8.0"/>
</dependencies>
```

and if I change "Builds for" to iOS 11 the error disappears.

The xml now contains

```
<dependencies>
    <deployment version="4352" identifier="iOS"/>
    <plugIn identifier="com.apple.InterfaceBuilder.IBCocoaTouchPlugin" version="15706"/>
    <capability name="Named colors" minToolsVersion="9.0"/>
    <capability name="documents saved in the Xcode 8 format" minToolsVersion="8.0"/>
</dependencies>
```

and, by the way, I use `prepare` and `compile` instead of `build`.

(I speculate this is a bug of omission where the `version` attribute is not set.)

**Comment by damsorian** [comment]
the same, any update?

**Comment by rudiminty** [comment]
I'm experiencing the same here, any feedback yet?

**Comment by lcsdms** [comment]
> @marcelmedina
> 
> ```
> error: Named colors do not work prior to iOS 11.0.
> ```
> 
> If you're using Cordova-iOS 6.x, the minimum supported iOS version is 11.0. You may need to remove a `deployment-target` preference from config.xml if it is set to a lower version.

Only after removing the tag from the config.xml it works in here. When using with value = "11" the problem still happens on the story board.

**Comment by lcsdms** [comment]
FYI, using cordova 6.x and setting the deployment-target to 11.0 also works:

`
    <preference name="deployment-target" value="11.0" />

`

**Comment by lukas2** [comment]
Unfortunately setting the deployment-target preference does not work in my setup. What does work is a hacky editing of the storyboard file. Basically i do this:

```
sed -i'.bak' 's/<deployment identifier=\"iOS\"\/>/<deployment version=\"4352\" identifier=\"iOS\"\/>/g' "path/to/my/CDVLaunchScreen.storyboard"
```

Perhaps the storyboard should be "parameterise-able" like this?

**Comment by XtoZee** [comment]
> FYI, using cordova 6.x and setting the deployment-target to 11.0 also works:
> 
> `<preference name="deployment-target" value="11.0" />`

Confirm that changing deployment-target from 9.0 to 11.0 helped me too!

**Comment by jasp255** [comment]
Try with previous version of ios library:

cordova platform rm ios
cordova platform add ios@5.1.1

**Comment by firstmanonthesun** [comment]
+1 to installing previous version of platform library, as per @jasp255

**Comment by Faksprod** [comment]
> Try with previous version of ios library:
> 
> cordova platform rm ios
> cordova platform add ios@5.1.1

Hi @jasp255 and @firstmanonthesun : I don't get why it seems to be a nice solution to downgrade your Cordova iOS platform library version. What I understood is that the Apple Store will now reject every app using `UIWebview` (which is the case for all iOS Cordova version before 6). So if you downgrade your Cordova project to `ios@5.1.1` how can you bypass this new Apple's restriction? Thanks!

**Comment by pschorey** [comment]
> @marcelmedina
> If you're using Cordova-iOS 6.x, the minimum supported iOS version is 11.0. You may need to remove a `deployment-target` preference from config.xml if it is set to a lower version.

Doesn't this mean anyone using iOS versions below 11.0 won't be able to install or update the app?  A significant portion of my app users are running iOS versions below 11 and I'd prefer not to lose them.  Is there another solution?

**Comment by breautek** [comment]
> Doesn't this mean anyone using iOS versions below 11.0 won't be able to install or update the app? 

Correct.

> Is there another solution?

You can stay on `cordova-ios@5.1.1`. You'd need to install a wkwebview plugin and add `<preference name="WKWebViewOnly" value="true" />` flag to your `config.xml` to make the app compliant with the `UIWebView` deprecation. More details can be found on [Cordova's Blog](https://cordova.apache.org/howto/2020/03/18/wkwebviewonly.html).

The primarily reason why `cordova-ios@6` requires min iOS 11.0 is because it uses the WKWebView url scheme feature to solve many CORS-related issues that occurs on WKWebView when using the `file://` protocol. This feature only exists in iOS 11+. You may also see these issues on `cordova-ios@5.1.1`, particularly if you use `XMLHttpRequest` on your local resources (which is a common pattern if you use the angular framework). If this is the case, you'll also need to use a third-party plugin `cordova-plugin-wkwebview-file-xhr@2.1.4` Note the plugin version, their latest version is only for `cordova-ios@6` and later.

This won't be a long term solution, as time goes on, changes will likely be required to support newer xcode versions, and these generally aren't backported to older versions. But this should work for awhile.

**Comment by JSeligsohn** [comment]
I'm having a similar issue:
I get a warning about the target version and an error about the CDVStoryboard. Setting the target preference in config.xml does not help me. I also can't manually change the target version in xCode because I'm building my app using CI. 
The specific warning seems to be coming from the Firebase plugin I'm using, but I'm not sure the source of the launch screen error. Any help would be appreciated.
```
....
warning: The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.0.99. (in target 'FirebaseRemoteConfig' from project 'Pods')
23:19:18
  ** ARCHIVE FAILED **
23:19:18
  
23:19:18
  
23:19:18
  The following build commands failed:
23:19:18
    CompileStoryboard /<my-path>/platforms/ios/{AppName}/CDVLaunchScreen.storyboard
23:19:18
  (1 failure)
23:19:18
  xcodebuild: Command failed with exit code 65
```

**Comment by breautek** [comment]
@JSeligsohn 

exit code 65 is a "generic" error that apple produces. There are a huge variety of reasons why Apple may return exit code 65. While you may have the same exit code, it's definitely not the same issue as the OP.

As the warning states in your output, you have your deployment target set to 8, which is lower than the supported versions by a dependency (`FirebaseRemoteConfig`) which supports versions between 9 and 14.

Additionally, `cordova-ios@6` requires at least version 11, so if you're using the latest version of `cordova-ios`, then you must set this to 11. Otherwise, you must set this to 9. See [config.xml docs](https://cordova.apache.org/docs/en/latest/config_ref/) for how to set the `deployment-target` preference.

If you require more assistance, please ask on our [slack](http://slack.cordova.io/). If you think you've found a bug with Cordova, then please open a new issue.

**Comment by breautek** [comment]
Closing this issue as the OP's issue had to deal with the change in build system where older cordova-ios versions had to use the legacy build system.

`cordova-ios@6` supports the new build system, so changing to legacy shouldn't be required anymore.

See the [SO accepted answer](https://stackoverflow.com/a/61857492/4685664) for more info.

**Comment by JSeligsohn** [comment]
@breautek Thank you for your prompt reply! You're right it's not the same error as the OP, but I noticed @marcelmedina had the same error as me.
This doesn't appear to be a specific bug with cordova-ios, you're right. I'm more seeking help with my problem (which appears to be related to cocoapods from a plugin setting their own minimum version). I'll try to reach out on slack.

**Comment by driiftkiing** [comment]
> > FYI, using cordova 6.x and setting the deployment-target to 11.0 also works:
> > `<preference name="deployment-target" value="11.0" />`
> 
> Confirm that changing deployment-target from 9.0 to 11.0 helped me too!

i could solve this by adding `.0` to the version (`11`)
```diff
- <preference name="deployment-target" value="11" />
+ <preference name="deployment-target" value="11.0" />
```
